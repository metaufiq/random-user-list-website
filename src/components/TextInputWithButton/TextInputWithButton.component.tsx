import { Fragment } from "react";
import Button from "../Button";
import TextInput from "../TextInput/TextInput.component";
import { Props } from "./TextInputWithButton.component.types";

const TextInputWithButton = ({label, htmlFor, id, onChange, type, buttonLabel, onPress, placeholder}: Props)=>{
  return(
    <div>
      <label htmlFor={htmlFor} className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-400">
        {label}
      </label>
      <div className="flex flex-row">
        <div className="mr-2 flex-auto">
          <TextInput 
            id={id} 
            placeholder={placeholder} 
            type={type} 
            onChange={onChange}
          />
        </div>
        <Button onClick={onPress}>
          {buttonLabel}
        </Button>
      </div>
    </div>

  )
}

export default TextInputWithButton;