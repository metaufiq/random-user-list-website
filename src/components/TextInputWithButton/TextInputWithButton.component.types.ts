import { ChangeEventHandler, HTMLInputTypeAttribute } from "react"

export type Option = {
  value: any,
  label: string,
}

export type Props = {
  placeholder: string,
  id?: string,
  type: HTMLInputTypeAttribute,
  label: string,
  buttonLabel: JSX.Element | string,
  htmlFor?: string,
  onChange: ChangeEventHandler<HTMLInputElement>,
  onPress: VoidFunction,
  value?: string | number | readonly string[] | undefined
}