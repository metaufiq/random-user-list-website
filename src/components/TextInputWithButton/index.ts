import TextInputWithButton from "./TextInputWithButton.component";

export default TextInputWithButton