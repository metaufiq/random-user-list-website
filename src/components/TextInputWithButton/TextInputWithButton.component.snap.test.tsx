import { assertSnapshots } from "../../../test";
import TextInputWithButton from ".";
import { Props } from "./TextInputWithButton.component.types";

describe('TextInputWithButton Snapshot', () => {
  const mockProps:Props = {
    onChange:()=>{},
    label: 'label',
    buttonLabel: 'button label',
    onPress: ()=>{},
    placeholder: 'placeholder',
    type: 'password'
  };

  const mockPropsWithOptionalProps: Props = {
    ...mockProps,
    htmlFor: 'htmlFor',
    id: 'id',
    value: 'value'
  }

  const configs = [
    {
      props: mockProps,
      desc: 'should render normal TextInputWithButton'
    },
    {
      props: mockPropsWithOptionalProps,
      desc: 'should render normal TextInputWithButton with optional props'
    }
  ];

  assertSnapshots(TextInputWithButton, configs);
});
