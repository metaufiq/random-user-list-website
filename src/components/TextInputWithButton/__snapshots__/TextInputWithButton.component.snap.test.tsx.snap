// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputWithButton Snapshot should render normal TextInputWithButton 1`] = `
<div>
  <label
    className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-400"
  >
    label
  </label>
  <div
    className="flex flex-row"
  >
    <div
      className="mr-2 flex-auto"
    >
      <input
        className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        onChange={[Function]}
        placeholder="placeholder"
        type="password"
      />
    </div>
    <button
      className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
      onClick={[Function]}
      type="button"
    >
      button label
    </button>
  </div>
</div>
`;

exports[`TextInputWithButton Snapshot should render normal TextInputWithButton with optional props 1`] = `
<div>
  <label
    className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-400"
    htmlFor="htmlFor"
  >
    label
  </label>
  <div
    className="flex flex-row"
  >
    <div
      className="mr-2 flex-auto"
    >
      <input
        className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        id="id"
        onChange={[Function]}
        placeholder="placeholder"
        type="password"
      />
    </div>
    <button
      className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
      onClick={[Function]}
      type="button"
    >
      button label
    </button>
  </div>
</div>
`;
